dspace-api/src/main/java/org/dspace/importer/external/crossref/CrossRefImportMetadataSourceServiceImpl.java

```diff
@@ -0,0 +1,332 @@
+    }
+
+    @Override
+    public ImportRecord getRecord(Query query) throws MetadataSourceException {
+        List<ImportRecord> records = null;
+        String id = getID(query.toString());
+        if (StringUtils.isNotBlank(id)) {
+            records = retry(new SearchByIdCallable(id));
+        } else {
+            records = retry(new SearchByIdCallable(query));
+        }
+        return records == null || records.isEmpty() ? null : records.get(0);
+    }
+
+    @Override
+    public Collection<ImportRecord> findMatchingRecords(Query query) throws MetadataSourceException {
+        String id = getID(query.toString());
+        if (StringUtils.isNotBlank(id)) {
+            return retry(new SearchByIdCallable(id));
+        }
+        return retry(new FindMatchingRecordCallable(query));
+    }
+
+
+    @Override
+    public Collection<ImportRecord> findMatchingRecords(Item item) throws MetadataSourceException {
+        throw new MethodNotFoundException("This method is not implemented for CrossRef");
+    }
+
+    public String getID(String query) {
+        if (DoiCheck.isDoi(query)) {
+            return "filter=doi:" + query;
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private class SearchByQueryCallable implements Callable<List<ImportRecord>> {
+
+        private Query query;
+
+        private SearchByQueryCallable(String queryString, Integer maxResult, Integer start) {
+            query = new Query();
+            query.addParameter("query", queryString);
+            query.addParameter("count", maxResult);
+            query.addParameter("start", start);
+        }
+
+        private SearchByQueryCallable(Query query) {
+            this.query = query;
+        }
+
+        @Override
+        public List<ImportRecord> call() throws Exception {
+            List<ImportRecord> results = new ArrayList<>();
+            Integer count = query.getParameterAsClass("count", Integer.class);
+            Integer start = query.getParameterAsClass("start", Integer.class);
+
+            URIBuilder uriBuilder = new URIBuilder(ENDPOINT_WORKS);
+            uriBuilder.addParameter("query", query.getParameterAsClass("query", String.class));
+            if (Objects.nonNull(count)) {
+                uriBuilder.addParameter("rows", count.toString());
+            }
+            if (Objects.nonNull(start)) {
+                uriBuilder.addParameter("offset", start.toString());
+            }
+
+            String response = liveImportClient.executeHttpGetRequest(1000, uriBuilder.toString(),
+                    new HashMap<String, String>());
+            ReadContext ctx = JsonPath.parse(response);
+            Object o = ctx.read("$.message.items");
+            if (o.getClass().isAssignableFrom(JSONArray.class)) {
+                JSONArray array = (JSONArray) o;
+                int size = array.size();
+                for (int index = 0; index < size; index++) {
+                    Gson gson = new Gson();
```

Comment:
I have tried to remove the json library as by your suggestion but it ends up with large conflict and refactoring needs (all the mapping). It would took not less than 12-16 hours to implement this refactoring. Please note that we have already performed a refactoring due to the Axiom removal that has increased the original estimation for this task, should we include also that or can we postpone that to a followup PR?

Reply:
@Micheleboychuk : These PRs cannot be merged as-is since GSon has been removed from the `main` branch entirely (i.e. #8259 was merged as no one complained about it).  Therefore, for these to be considered for 7.3 they must be updated.

Please always keep in mind that your feedback is welcome on others PRs.  If you notice impacts like these, speak up.  Gson was removed from the codebase as it was used in only one place in our codebase before these PRs... and the version of Gson we were using had known security vulnerabilities.

I can update the estimates for these PRs as needed if you want to continue to have them submitted for 7.3. I'd need to know what the new total estimate is. Please add it to the original ticket.